examples: Add Examples 5 & 6 and maintainer info request (#3601)
perma-id/w3id.org: examples/README.md

  * [Example 1: Minimalist (grouping)](#example-1-minimalist-grouping)
  * [Example 2: Supporting multiple media types (MIME types)](#example-2-supporting-multiple-media-types-mime-types)
  * [Example 3: Dealing with query string](#example-3-dealing-with-query-string)
  * [Example 4: Publish vocabularies with W3ID](#example-4-publish-vocabularies-with-w3id)
  * [Example 5: Version-aware URIs of ontologies](#example-5-version-aware-uris-of-ontologies)
* [README.md](#readmemd)

## .htaccess

> `.htaccess` file is the key for the working of URL redirection service of W3ID.

Reviewer: ```suggestion
`.htaccess` file is the key to the working of URL redirection service of W3ID.
```

Without it, redirection cannot be done.

> ### What is `.htaccess?`

Reviewer: ```suggestion
### What is `.htaccess`?
```

web pages and files), and more. The 'dot' (period or full stop) before the
file name makes it a hidden file in Unix-based environments. 

> In W3ID context, it is used primarily for URL redirection. The `.htaccess`
> file is where you can put URL rewriting rules in. A set of URL rewriting rules
> will work together and effectively made URL redirection happen.

Reviewer: ```suggestion
In the W3ID context, `.htaccess` is used primarily for URL redirection. This
file is where you can put URL rewriting rules. A set of URL rewriting rules
will work together and effectively make URL redirection happen.
```

### Put ID info and maintainer info inside .htaccess

> Maintainers of an ID are required to put a breif ID info and a maintainer info
> in the comment (lines staring with `#` character) of a `.htaccess` file.
> 
> Having contact info and GitHub username help the overall maintainance process
> easier.

Reviewer: ```suggestion
Maintainers of an ID are requested to put brief ID info and maintainer info
in the comments (lines staring with `#` character) of an `.htaccess` file.

Including contact info and GitHub usernames helps ease the overall
maintenance process.
```

### Example 2: Supporting multiple media types (MIME types)

A single resource, located by the same URL, can have multiple representations.
For example, an image can can be represented in both GIF and PNG formats.

> A web server can be configured to return different media type or file format
> depends on the client's request or capability. We call this mechanism a

Reviewer: ```suggestion
A web server can be configured to return a different media type or file format
depending on the client's request or capability. We call this mechanism
```

Where *CondPattern* is a regular expresssion to match pattern in *TestString*.

> In this case, *TestString* is `%{HTTP_ACCEPT}` which its value is taken from

Reviewer: ```suggestion
In this case, *TestString* is `%{HTTP_ACCEPT}` which value is taken from
```

an `Accept` field in the HTTP request header. The `Accept` field can be a
string like this:

```HTTP
Accept: text/html, application/xhtml+xml, application/xml, image/webp
```

> Each media type will be presented, separated by a comma. With that, we can use
> *CondPattern* to matches media types in this string.

Reviewer: ```suggestion
Each media type will be presented, separated by a comma. We can then use
*CondPattern* to match media types in this string.
```

### Example 3: Dealing with query string

Everything after the question mark (`?`) in the URL, but not that `?` itself,
is a query string.

For example, for the URL `https://en.wikipedia.org/w/index.php?title=Web`,
the query string is `title=Web`.

As the query string is not included in the string that the *Pattern* of
*RewriteRule* will compared against, you cannot use *Pattern* to match them.

> To find pattern in the query string, use `%{QUERY_STRING}` as a *TestString*

Reviewer: ```suggestion
To find a pattern in the query string, use `%{QUERY_STRING}` as a *TestString*
```

in *RewriteCond*.

> As an example, if you like to redirect the URL

Reviewer: ```suggestion
As an example, if you would like to redirect the URL
```

> In this example, the *CondPattern* `(.*)` in *RewriteCond* will match every characters in the query string (`a=1&b=2`) and put it in group one. This group one (`%1`) is later used in the *Substitution* of *RewriteRule*.

Reviewer: ```suggestion
In this example, the *CondPattern* `(.*)` in *RewriteCond* will match every character in the query string (`a=1&b=2`) and put it in group one. This group one (`%1`) is used later in the *Substitution* of *RewriteRule*.
```

Note that the special character to recall groups from *CondPattern* of *RewriteCond* is `%` (unlike the special character to recall groups from *Pattern* of *RewriteRule*, which is `$`).

The character `?` at the end of the *Substitution* of *RewriteRule* tells the server not to pass the query string to the final URL after rewrite.

### Example 4: Publish vocabularies with W3ID

If you plan to publish a vocabulary/ontology with W3ID,

> see [/example/.htaccess](https://github.com/perma-id/w3id.org/blob/master/example/.htaccess)

Reviewer: ```suggestion
see [`/example/.htaccess`](https://github.com/perma-id/w3id.org/blob/master/example/.htaccess)
```

and https://w3id.org/example/.

### Example 5: Version-aware URIs of ontologies

> See [/OWLunit/.htaccess](https://github.com/perma-id/w3id.org/blob/master/OWLunit/.htaccess).

Reviewer: ```suggestion
See [`/OWLunit/.htaccess`](https://github.com/perma-id/w3id.org/blob/master/OWLunit/.htaccess).
```

## README.md

> Each ID hosted on W3ID is required to have a file named `README.md` containing
> an information about the ID itself and an information about the maintainer(s).
> This can be more elaborate than the information inside `.htaccess`.
> 
> The `.md` file extension at the end of the file name indicates that the file
> uses Markdown markup language for text formatting. You can have tables and
> images as well, if needed.
> 
> GitHub will automatically display the content of a `README.md` to repository
> visitors.
> 
> An example of a good README file:
> [w3id.org/dggs/README.md](https://github.com/perma-id/w3id.org/blob/master/dggs/README.md)

Reviewer: ```suggestion
Each ID hosted on W3ID is required to have a file named `README.md` containing
information about the ID itself and information about the maintainer(s).
This can be more elaborate than the information inside `.htaccess`.

The `.md` file extension at the end of the file name indicates that the file
uses Markdown markup language for text formatting. You can have tables and
images as well, if needed.

GitHub will automatically display the content of a `README.md` to repository
visitors.

An example of a good README file:
[`w3id.org/dggs/README.md`](https://github.com/perma-id/w3id.org/blob/master/dggs/README.md)
```

See [`/OWLunit/.htaccess`](https://github.com/perma-id/w3id.org/blob/master/OWLunit/.htaccess).

### Example 6: Redirection based on a file extension in the URL

Check for pattern of a file extension (like `.json`, `.png`, `.rdf`, `.html`)
in the requested URL, and rewrite the URL accordingly.

This technique can compliments `RewriteCond %{HTTP_ACCEPT}` rules (explained

> in [Example 2](#example-2-supporting-multiple-media-types-mime-types)).

Reviewer: ```suggestion
Check for the pattern of a file extension (like `.json`, `.png`, `.rdf`, `.html`)
in the requested URL, and rewrite the URL accordingly.

This technique can complement `RewriteCond %{HTTP_ACCEPT}` rules (explained
in [Example 2](#example-2-supporting-multiple-media-types-mime-types)).
```
